tests/qt_types_standalone/rust/src/qtimezone.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx_qt_lib::{QByteArray, QTimeZone};
+
+#[cxx::bridge]
+mod qtimezone_cxx {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qtimezone.h");
+        type QTimeZone = cxx_qt_lib::QTimeZone;
+    }
+
+    extern "Rust" {
+        fn construct_qtimezone() -> QTimeZone;
+        fn read_qtimezone(t: &QTimeZone) -> bool;
+        fn clone_qtimezone(t: &QTimeZone) -> QTimeZone;
+    }
+}
+
+fn construct_qtimezone() -> QTimeZone {
+    QTimeZone::from_iana(&QByteArray::from("Europe/London"))
```

Comment:
valgrind is failing somewhere here...

Reply:
Appears to be related to the inner QSharedDataPointer<T> being set by a non default QTimeZone (when default it is nullptr), then if that QTimeZone is given to C++ it leaks.

Reply:
If a non default QTimeZone is constructed in Rust and given to C++, if C++ then calls `tz.~QTimeZone()` then the memory is freed correctly :thinking:

Reply:
Moved into #420